### PR TITLE
doc: porting: add note about CDC ACM UART configuration

### PR DESCRIPTION
**Pull request by jfischer-no**
Follow up on commit 0cae816b27f67f0490a90d7c9e7df2d722dfdd97
("boards: add common configuration for CDC ACM UART")

### CODE REVIEW

**Review by nordicjm — COMMENTED** [review]
`The fact that it was merged in this state indicates serious problems in the board PR review and merge process.`

You what?

**Comment by nordicjm** [comment]
@mtfurlan can you check and see if the changes here are OK?

**Comment by nordicjm** [comment]
Right and how is that a "serious problems in the board PR review and merge process"? And notwithstanding that the submitter is free to use USB instead of a UART if they want:

> If available, configure and enable a serial output for the console using the zephyr,console chosen node in the devicetree.

**Comment by jfischer-no** [comment]
> Right and how is that a "serious problems in the board PR review and merge process"? And notwithstanding that the submitter is free to use USB instead of a UART if they want:
> 
> > If available, configure and enable a serial output for the console using the zephyr,console chosen node in the devicetree.

Surely it is not up to the submitter to use the CDC ACM UART as the default instead of UART controller when UART controller is available to the host and has never been in the project. It is an unwritten rule, and you should be familiar with it. It also makes no sense, as it consumes more resources for no reason.

**Comment by jfischer-no** [comment]
> It's what the submitter determines is better suite if the UART works fine then sure, use the UART, I don't have that board and neither do you so neither of us can comment on why USB CDC was used instead.A board might have 3 UARTs and need to be manually wired up because there isn't one routed to e.g. USB, it's up to the submitter to determine which one of those to pick as the console, likewise a UART on a board might have limitations (e.g. bad throughput, lack of flow control, low maximum baud rate) and they might decide to pick another, that's up to them to decide. No there is no such unwritten rule, UART is better in that you can use it for tests with twister but that doesn't mean it's the requirement

No. We need consistent behavior and configuration across all development boards for these trivial things. Submitter are free to do whatever they want in their downstream repository. Here they have to follow the rules (also unwritten) about how to configure the boards and how to submit PRs. 
Obviously if there were a need to solder or jump the UART connection, or a hardware problem that forced to use CDC ACM UART to be used by default, it would be documented and mentioned in the commit message (30fa48558e1e8070b38517e31b37080f45068902). This is not the case for this board. From the DWM3001CDK manual, "An on-board J-Link debugger provides SWD and UART interfaces to the DWM3001C.", where it is connected to the RX/TX pins via J14/J15, with the default state "Soldered". I do not think it is the board reviewer's duty to look in the manuals, but anything that does not use a real UART controller on a development board should be justified.

**Comment by jfischer-no** [comment]
> An "unwritten rule" is also known as "not actually a rule"
> 
> In this case I had copied a different board config that did the usb cdc acm and gotten it working, and I never went back to figure out why uart0 didn't work (it works fine, turns out)

Really, I am just curious, how did you get the idea to just copy some arbitrary Kconfig file with some CDC ACM settings, when the vast majority of development boards use the UART connected to the debug adapter, and the very similar board decawave_dwm1001 uses the UART connected to the similar debug adapter, [and you even copied the files from this board](https://github.com/zephyrproject-rtos/zephyr/pull/79019/files#diff-c695ca6a09a6d8bef1a8a6c8cab5f911a7b2d0a21b77f7145c7136d11713bfd1R2), and the manual of the development board states that there is a UART connected to the debug probe?
 
> I verified the uart0 works fine in this PR, which is great this is a much simpler board config
> 
> I don't see how to test the cdc acm stuff using the configs in boards/common/usb/cdc_acm_serial without making a new board, `west build` accepts the dtsi with `--extra-dtc-overlay`, but I don't see how to apply the defconfig Is there a simple way to tell west to do it, and do we even care about testing that?

Sorry, no idea what you mean.

> I think this would make the decawave_dwm3001cdk documentation less misleading now that I understand it better(be more clear about the usb ports, and copy the flashing instructions from decawave_dwm1001_dev instead of the two port stuff I wrote when I thought that cdc acm on the second usb port was the only way to get serial output)

Please open a PR to improve decawave_dwm3001cdk documentation.

**Comment by mtfurlan** [comment]
> how did you get the idea to just copy some arbitrary Kconfig file with some CDC ACM settings

I tried to get uart working, something was wrong, I looked at the documentation that said that there was a usb port connected to the nrf52833 so I tried that and it worked

> Sorry, no idea what you mean.

I figured it out, so nevermind.
CDC ACM on the second usb port still works with this PR

> Please open a PR to improve decawave_dwm3001cdk documentation.

The documentation changes should go with the changes you have in the commit [boards: decawave_dwm3001cdk: remove CDC ACM configuration](https://github.com/zephyrproject-rtos/zephyr/pull/83107/commits/72aaa77660ab73afdd0d391b0e5cdcdedb9e1705), as you fixing the cdc acm stuff means the instructions tell you to do overly complicated stuff that isn't necessary anymore

**Review by mtfurlan — APPROVED** [review]

File: doc/hardware/porting/board_porting.rst [hunk 1]
```diff
@@ -649,6 +649,15 @@ while porting.
...
 - If available, configure and enable a serial output for the console
   using the ``zephyr,console`` chosen node in the devicetree.
+  Development boards with a built-in debug adapter or USB-to-UART adapter should
+  by default configure and use the UART controller connected to that adapter.
+  For boards like :ref:`nRF52840 Dongle <nrf52840dongle_nrf52840>`, that do not
+  have a debug adapter, but a USB device controller, there is a common
+  :zephyr_file:`Kconfig file <boards/common/usb/Kconfig.cdc_acm_serial.defconfig>`
+  that must be included in the board's Kconfig.defconfig file and
+  :zephyr_file:`devicetree file <boards/common/usb/cdc_acm_serial.dtsi>`
+  that must be included if the board's devicetree, if the board want to use the
```
Comment (carlescufi) [inline doc/hardware/porting/board_porting.rst:659]:
```suggestion
  that must be included if the board's Devicetree, if the board want to use the
```

**Review by carlescufi — APPROVED** [review]
LGTM, minor typo.

**Review by fabiobaltieri — APPROVED** [review]

**Comment by fabiobaltieri** [comment]
Hey I think the `Kconfig.cdc_acm_serial.defconfig` may be worth a line in the release notes. Not for this PR of course.

**Review by henrikbrixandersen — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Pull request merged**